js/store.js

```diff
@@ -157,6 +306,37 @@ const store = {
     }
   },
 
+  // Checks if third party is on whitelist for first party,
+  // and if the whitelisted third party should be unwhitelisted.
+  whiteListCheck(origin, target, firstParty, thirdParty) {
+    let newThirdParty = false;
+    if (!('isVisible' in thirdParty) || thirdParty['isVisible']) {
+      // third party needs a whitelist check
+      if (this.onWhiteList(origin, target) && !('isVisible' in thirdParty)) {
+        // third party is whitelisted for this first party
+        // ignore the link for now
+        thirdParty['isVisible'] = false;
+        thirdParty['whiteListedFirstParty'] = origin;
+      } else {
+        // third party is not whitelisted for this first party
+        firstParty['thirdPartyHostnames'].push(target);
+        newThirdParty = true;
+      }
+    } else if (!(this.onWhiteList(origin, target))) {
+      // third party was previously whitelisted, and its
+      // previously ignored link with the owning first party
+      // should be added now that it links to a first party
+      // for which it is not whitelisted
+      thirdParty['isVisible'] = true;
+      const whiteListedFirstParty = thirdParty['whiteListedFirstParty'];
+      this._websites[whiteListedFirstParty]['thirdPartyHostnames']
+        .push(target);
+      firstParty['thirdPartyHostnames'].push(target);
+      newThirdParty = true;
+    }
+    return [firstParty, thirdParty, newThirdParty];
```

Comment:
I initially tried to do that; but then I get a `Parsing error: Unexpected token =` error where the `this.whiteListCheck` is called:

```javascript
      { firstParty, thirdParty, newThirdParty }
        = this.whiteListCheck(origin, target, firstParty, thirdParty);
```
Any ideas why? Also why are argument return orders more consistent using an object instead of an array? I guess this isn't really an object since it doesn't have key-value pairs...